src/odemis/driver/piezomotor.py

```diff
@@ -523,16 +524,22 @@ def _check_axis_error(self, axis_id: int, status: list):
         elif status[1] & 4:
             raise PMDError(6, "External limit reached, detected on axis %s." % axis_id)
 
-    def _updatePosition(self):
+    def _updatePosition(self, axes: Optional[set] = None):
         """
         Update the position VA.
+        :param axes: names of the axes to update or None if all should be updated
         """
         pos = {}
-        for axname, axis in self._axis_map.items():
-            # TODO: if not in closed-loop, it's probably because there is no encoder, so we need a different way
-            pos[axname] = self.getEncoderPosition(axis)
+        for axname in self._axis_map:
+            if axes is None or axname in axes:
+                # TODO: if not in closed-loop, it's probably because there is no encoder, so we need a different way
+                pos[axname] = self.getEncoderPosition(axname)
         logging.debug("Reporting new position at %s", pos)
         pos = self._applyInversion(pos)
+        if axes is not None:
+            pos_full = self.position.value
```

Comment:
You need to make a copy (of the dict). Otherwise, afterwards `self.position.value` already contains the new position, and when setting the "new" value, the VA will not notify the subscribers.